src/Client/Core/DurableTaskClient.cs

```diff
@@ -464,15 +465,61 @@ public virtual Task RewindInstanceAsync(
         string instanceId,
         string reason,
         CancellationToken cancellation = default)
-        => throw new NotSupportedException($"{this.GetType()} does not support orchestration rewind.");
-
-    // TODO: Create task hub
-
-    // TODO: Delete task hub
-
-    /// <summary>
-    /// Disposes any unmanaged resources associated with this <see cref="DurableTaskClient"/>.
-    /// </summary>
-    /// <returns>A <see cref="ValueTask"/> that completes when the disposal completes.</returns>
-    public abstract ValueTask DisposeAsync();
-}
+        => throw new NotSupportedException($"{this.GetType()} does not support orchestration rewind.");
+
+    /// <summary>
+    /// Lists orchestration instance IDs filtered by completed time.
+    /// </summary>
+    /// <param name="runtimeStatus">The runtime statuses to filter by.</param>
+    /// <param name="completedTimeFrom">The start time for completed time filter (inclusive).</param>
+    /// <param name="completedTimeTo">The end time for completed time filter (inclusive).</param>
+    /// <param name="pageSize">The page size for pagination.</param>
+    /// <param name="lastInstanceKey">The last fetched instance key.</param>
+    /// <param name="cancellation">The cancellation token.</param>
+    /// <returns>A page of instance IDs with continuation token.</returns>
+    public virtual Task<Page<string>> ListInstanceIdsAsync(

```

Comment:
Instead of this generic name, if this is designed to always filter by completed time (as per the comment), we should update the method name to reflect that.

Or, if the comment is wrong, update that to have this be more generic. It feels more like a query than a list with all the filters.

Reply:
Also, how does this differ from `GetAllInstancesAsync`? Is it just the query/filter ability?

Reply:
as below